lib/services.js

```diff
@@ -746,7 +750,13 @@ const services = {
             'content-md5': contentMD5,
             'content-length': size,
         };
-        metadata.putObjectMD(mpuBucketName, partKey, omVal, {}, log, err => {
+
+        const params = {};
+        if (overheadField) {
+            params.overheadField = overheadField;
+        }
+
+        metadata.putObjectMD(mpuBucketName, partKey, omVal, params, log, err => {
```

Comment:
I also have a side question, I am trying to understand the context of these changes (I found nothing on Jira), why do we need to pass additional arguments as a new parameter while we have, here, for example, already the `omVal` variable that contains the content-length? Is it because from metadata, we do not know what are the expected headers for each API?

Reply:
The high level reason is that some operations don't leave enough context in the oplog for some of the things we want to do with it. For example if a non-versioned object is deleted we can see what key, but the content-lenth or who owned to object is lost.
This param is used to ask Metadata to attach the "previous" value of the field to the oplog entry to provide this missing context.

Specifically on passing the fields we have 2 things going on:
1. Cloudserver backend ops are async and if it were to pass the data directly there is a chance that the data could change if there was a concurrent write to the same key that was processed first.
2. Metadata doesn't know mush about the structure of object metadata.

To work around 1 we need to push the work to Metadata where requests are ordered and we can make atomic guarantees about the "previous" value.
Since 2 is a nice property to maintain it we pass only the fields we want extracted and then have a generic  mechanism in Metadata do the retrieval. 

The full design is over here https://github.com/scality/citadel/blob/development/1.0/docs/design/raft-log-overhead.md

Reply:
Thank you very much for the detailed explanation, I read the design and now it makes sense :smile: